packages/rancher-monitoring/templates/crd-template/templates/jobs.yaml

```diff
@@ -20,28 +20,28 @@ spec:
         runAsNonRoot: true
         runAsUser: 1000
       initContainers:
-        - name: delete-crds
+        - name: set-preserve-unknown-fields-false
           image: {{ template "system_default_registry" . }}{{ .Values.image.repository }}:{{ .Values.image.tag }}
           imagePullPolicy: IfNotPresent
           command:
-          - /bin/kubectl
-          - delete
-          - --ignore-not-found=true
-          - -f
-          - /etc/config/crd-manifest.yaml
-          volumeMounts:
-            - name: crd-manifest
-              readOnly: true
-              mountPath: /etc/config
+          - /bin/sh
+          - -c
+          - >
+            {{- range $path, $_ :=  (.Files.Glob  "crd-manifest/**.yaml") }}
+            {{- $crd := get (get ($.Files.Get $path | fromYaml) "metadata") "name" }}
+            if [[ -n "$(kubectl get crd {{ $crd }} -o jsonpath='{.spec.preserveUnknownFields}')" ]]; then
```

Comment:
This is a best effort operation (hence the `|| true`) that should be done only if the CRD exists and has `.spec.preserveUnknownFields` set to some value. 

If the CRD does not exist or if `.spec.preserveUnknownFields` is unset, this bash script will skip since the `kubectl get` command will return nothing to the stdout, but it will still print the error in finding the resource since the kubectl get command prints to stderr.